test/type-error.js

```diff
@@ -0,0 +1,105 @@
+import test from 'ava';
+import avaRuleTester from 'eslint-ava-rule-tester';
+import rule from '../rules/type-error';
+
+const ruleTester = avaRuleTester(test, {
+	env: {
+		es6: true
+	}
+});
+
+const errors = [{
+	ruleId: 'type-error',
+	message: '`new Error()` is too unspecific for a typecheck, use `new TypeError()` instead.'
+}];
+
+ruleTester.run('type-error', rule, {
+	valid: [
+		`if (Array.isArray(foo) || ArrayBuffer.isView(foo)) {
```

Comment:
Sorry for the addionnal work :sweat_smile: 
But this will most definitely pay off :wink: